resources/examples/host/simpleResolver/simpleResolver.py

```diff
@@ -0,0 +1,178 @@
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASISc
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+#
+"""
+A simple CLI that demonstrates resolving an Entity Reference though
+the default OpenAssetIO manager.
+
+Simply supply a set of traits and an entity reference, and the resulting
+data will be output in JSON form.
+"""
+import argparse
+import json
+import sys
+
+
+from openassetio import SpecificationBase
+from openassetio.hostApi import HostInterface, ManagerFactory
+from openassetio.log import ConsoleLogger, SeverityFilter
+from openassetio.pluginSystem import PythonPluginSystemManagerImplementationFactory
+
+
+# pylint: disable=missing-function-docstring,no-self-use,invalid-name
+
+
+#
+## OpenAssetIO required host classes
+#
+
+
+class TestHost(HostInterface):
+    """
+    A minimal host interface implementation. This identifiers the
+    calling tool or application to the API and downstream Manager.
+    """
+
+    def identifier(self):
+        return "default.manager.demo.host"
+
+    def displayName(self):
+        return "Default Manager Demo Host"
+
+
+class CLILocale(SpecificationBase):
+    """
+    A minimal locale that defines a CLI environment.
+    Note: In the future, these common starting points will be moved to a
+    shared repository to avoid inconsistencies/duplication.
+    """
+
+    kTraitSet = {"cli", "demo"}
+
+
+#
+## Batch result callbacks
+#
+# The API is batch-centric for entity related calls. The general
+# approach supplies a list of Entity References to a method, which then
+# calls the success or failure callback for each element of that list
+# with the response from the manager. In the future we plan to provide
+# additional conveniences for single entity tasks, or 'list in list out'
+# use cases.
+
+
+def print_traits_data(_, data):
+    """
+    A callback function that will print out the supplied entity
+    TraitsData. As we only deal with a single result, we ignore the
+    index.
+    """
+    as_dict = {
+        trait_id: {
+            property_key: data.getTraitProperty(trait_id, property_key)
+            for property_key in data.traitPropertyKeys(trait_id)
+        }
+        for trait_id in data.traitSet()
+    }
+    print(json.dumps(as_dict))
+
+
+def fail_with_message(_, batch_element_error):
+    """
+    A callback function that will exit with an appropriate message and
+    error code As we only deal with a single result, we ignore the
+    index.
+    """
+    sys.stderr.write(f"ERROR: {batch_element_error.message}\n")
+    sys.exit(int(batch_element_error.code))
+
+
+#
+# main
+#
+
+
+def create_argparser():
+    """
+    Creates an argparse for the tool.
+    """
+    parser = argparse.ArgumentParser()
+    parser.add_argument(
```

Comment:
[consider][minor] This argument could allow multiple inputs, rather than being comma-separated, e.g. `nargs=+` (and/or `action='append'`).  Happy to leave as comma-separated, just wanted to make sure all options were considered.

Reply:
Added #638